### PR TITLE
feat(sdk-coin-sui): update balance querying to handle fundsInAddressBalance field

### PR DESCRIPTION
**Pull request by abhishekagrawal080**
TICKET: CSHLD-407

### CODE REVIEW

**Review by veetragjain — DISMISSED** [review]

**Comment by lokesh-bitgo** [comment]
@claude Please do a thorough review of this PR.

Focus especially on:
- any bugs or incorrect logic in the changes
- whether these changes could impact existing flows or cause regressions
- whether the implementation is aligned with the requirement/design
- whether the changes are correct and complete
- missing edge cases, validation gaps, and test coverage gaps

Please call out concrete issues clearly and separate them by severity if possible.

**Comment by Ranjna-G** [comment]
@abhishekagrawal080 Is there any staking related changes that's required?

File: modules/sdk-coin-sui/src/lib/transferBuilder.ts [hunk 1]
```diff
@@ -148,6 +245,37 @@ export class TransferBuilder extends TransactionBuilder<TransferProgrammableTran
...
+      // Sender's funds come entirely from address balance; gas is paid by the sponsor.
+      // withdrawal(fundsInAddressBalance) → redeem_funds → Coin<SUI> → SplitCoins → TransferObjects
+      const [addrCoin] = programmableTxBuilder.moveCall({
+        target: '0x2::coin::redeem_funds',
+        typeArguments: ['0x2::sui::SUI'],
+        arguments: [programmableTxBuilder.withdrawal({ amount: BigInt(this._fundsInAddressBalance.toFixed()) })],
+      });
+      this._recipients.forEach((recipient) => {
+        const splitObject = programmableTxBuilder.splitCoins(addrCoin, [
+          programmableTxBuilder.pure(Number(recipient.amount)),
```
Comment (Ranjna-G) [inline modules/sdk-coin-sui/src/lib/transferBuilder.ts:262]: Shouldn't this be BigInt?

File: modules/sdk-coin-sui/src/lib/transactionBuilder.ts [hunk 2]
```diff
@@ -149,7 +159,15 @@ export abstract class TransactionBuilder<T = SuiProgrammableTransaction> extends
...
 
   validateGasPayment(payments: SuiObjectRef[]): void {
-    assert(payments && payments.length > 0, new BuildTransactionError('gas payment is required before building'));
+    assert(
+      payments !== undefined && payments !== null,
+      new BuildTransactionError('gas payment is required before building')
+    );
+    // Empty array is valid: Sui allows setGasPayment([]) when gas is paid from
+    // address balance (sender or fee payer has sufficient fundsInAddressBalance).
+    if (payments.length === 0) {
```
Comment (Ranjna-G) [inline modules/sdk-coin-sui/src/lib/transactionBuilder.ts:168]: Can we check if fundsInAddressBalance is non-zero first, and skip this check entirely if it is? That way, for object-only transactions where there's no address balance involved, we avoid unnecessarily hitting the node and getting an error.

File: modules/sdk-coin-sui/src/lib/transferBuilder.ts (hunk 1 above)
Comment (abhishekagrawal080) [reply modules/sdk-coin-sui/src/lib/transferBuilder.ts:262]: it was always like this  programmableTxBuilder.pure(Number(req.amount)), https://github.com/BitGo/BitGoJS/blob/4aa1ac8f3ccf86753f841eafdb1439edb3401ca9/modules/sdk-coin-sui/src/lib/walrusStakingBuilder.ts#L208

**Review by Ranjna-G — APPROVED** [review]

**Review by alextse-bg — APPROVED** [review]
wallets changes lgtm

**Review by lokesh-bitgo — APPROVED** [review]
WP Changes looks good
